### ISSUE TITLE
Dashboard: Telemetry; A11y improvements

### ISSUE THREAD

**Issue by carloskelly13** [GoogleForCreators/web-stories-wp#4627]
## Feature Description

1. The dismiss button should announce what it is dismissing.
2. When the telemetry checkbox becomes checked focus shifts to section container, it should remain on the checkbox.
3. The learn more link should announce what it is a link to.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by csossi** [comment]
Verified in QA
